src/main/java/net/sf/jabref/logic/autocompleter/AutoCompleterFactory.java

```diff
@@ -44,17 +44,20 @@ public AutoCompleterFactory(AutoCompletePreferences preferences, JournalAbbrevia
 
         if (InternalBibtexFields.getFieldExtras(fieldName).contains(FieldProperties.PERSON_NAMES)) {
             return new NameFieldAutoCompleter(fieldName, preferences);
-        } else if (FieldName.CROSSREF.equals(fieldName)) {
+        } else if (InternalBibtexFields.getFieldExtras(fieldName).contains(FieldProperties.SINGLE_ENTRY_LINK)) {
             return new BibtexKeyAutoCompleter(preferences);
-        } else if (FieldName.JOURNAL.equals(fieldName) || FieldName.PUBLISHER.equals(fieldName)) {
+        } else if (InternalBibtexFields.getFieldExtras(fieldName).contains(FieldProperties.JOURNAL_NAME)
+                || FieldName.PUBLISHER.equals(fieldName)) {
             return new JournalAutoCompleter(fieldName, preferences, abbreviationLoader);
         } else {
             return new DefaultAutoCompleter(fieldName, preferences);
         }
     }
 
     public AutoCompleter<String> getPersonAutoCompleter() {
-        return new NameFieldAutoCompleter(Arrays.asList(FieldName.AUTHOR, FieldName.EDITOR), true, preferences);
+        return new NameFieldAutoCompleter(InternalBibtexFields.getAllPublicFieldNames().stream().filter(
```

Comment:
I was also considering that. Maybe even introduce a constant list as there are for some other properties. Advantage: no need for two long lines of code. Disadvantages: not user configurable (if we ever want that, easy to change though) and, if we add a method, there may be quite a few special methods. Which on the other hand may be better than quite a few two line statements...

I might go for the constant list as there are probably more fields that should count as journal names and then we might include them in one step as well.